src/olympia/lib/git.py

```diff
@@ -164,14 +164,13 @@ def extract_and_commit_from_file_obj(cls, file_obj, channel, author=None):
 
         branch = repo.find_or_create_branch(BRANCHES[channel])
 
-        # Create a temporary worktree that we can use to unpack the zip
+        # Create a temporary worktree that we can use to unpack the extension
         # without disturbing the current git workdir since it creates a new
         # temporary directory where we extract to.
         with TemporaryWorktree(repo.git_repository) as worktree:
-            # Now extract the zip to the workdir
-            zip_file = SafeZip(file_obj.current_file_path, force_fsync=True)
-            zip_file.extract_to_dest(worktree.path)
-
+            # Now extract the extension to the workdir
+            extract_extension_to_dest(
+                file_obj.current_file_path, force_fsync=True)
```

Comment:
It may be worth adding a unit test for `extract_extension_to_dest()` so that all codepaths are covered - I don't care much about it if you end up just testing this as part of another test in a later PR, though.

Reply:
Added a handful of tests in 67bea048ca

Reply:
And ceedec7